File: Survey/Models/Questionario.cs
using Survey.DAL;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Survey.Models
{
    internal class Questionario
    {
        private int _id;
        private string _nome;
        private DateTime _inicio;
        private DateTime _fim;
        private string _msgFeedback;
        private string _guid;
        private string _imagem;
        private int _usuarioId;
        private Usuario _usuario;
        private List<Pergunta> _perguntas;

        internal int Id
        {
            get
            {
                return _id;
            }

            set
            {
                _id = value;
            }
        }

        internal string Nome
        {
            get
            {
                return _nome;
            }

            set
            {
                _nome = value;
            }
        }

        internal DateTime Inicio
        {
            get
            {
                return _inicio;
            }

            set
            {
                _inicio = value;
            }
        }

        internal DateTime Fim
        {
            get
            {
                return _fim;
            }

            set
            {
                _fim = value;
            }
        }

        internal string MsgFeedback
        {
            get
            {
                return _msgFeedback;
            }

            set
            {
                _msgFeedback = value;
            }
        }

        internal string Guid
        {
            get
            {
                return _guid;
            }

            set
            {
                _guid = value;
            }
        }

        internal string Imagem
        {
            get
            {
                return _imagem;
            }

            set
            {
                _imagem = value;
            }
        }

        internal int UsuarioId
        {
            get
            {
                return _usuarioId;
            }

            set
            {
                _usuarioId = value;
            }
        }

        internal Usuario Usuario
        {
            get
            {
                return _usuario;
            }

            set
            {
                _usuario = value;
            }
        }

        internal List<Pergunta> Perguntas
        {
            get
            {
                return _perguntas;
            }

            set
            {
                _perguntas = value;
            }
        }

        internal List<Questionario> ObterPorUsuario(int id)
        {
            if (id > 0)
                return new QuestionarioDAO().ObterPorUsuario(id);
            else
                return null;
        }

        internal List<Questionario> ObterPorPalavraChave(string palavra, int idUsuario)
        {
            if (palavra.Length > 0)
                return new QuestionarioDAO().ObterPorPalavraChave(palavra, idUsuario);
            else
                return null;
        }

        internal Questionario Obter(int id, int idUsuario)
        {
            if (id > 0)
                return new QuestionarioDAO().Obter(id, idUsuario);
            else
                return null;
        }

        internal int Gravar()
        {
            if (_id >= 0 && _nome.Length > 0 && _inicio > DateTime.MinValue && _fim > DateTime.MinValue &&
                _guid.Length > 0 && _usuarioId > 0)
                return new QuestionarioDAO().Gravar(this);
            else
                return -10;
        }
        internal Questionario BuscarQuestionario(string guid)
        {
            return new QuestionarioDAO().BuscarQuestionario(guid);
        }

        internal int Excluir(int id)
        {
            if (id > 0)
                return new QuestionarioDAO().Excluir(id);
            else
                return -10;
        }
    }
}

File: SurveyWeb/wwwroot/js/Views/Questionario.js
function PreencherTabela(dados) {
    var txt = '<thead>\
            <tr>\
                <th>#ID</th>\
                <th>Título</th>\
                <th>Início</th>\
                <th>Fim</th>\
                <th>Guid</th>\
                <th>...</th>\
            </tr>\
        </thead >\
        <tbody>';
    $.each(dados, function () {
        txt += '<tr><td>' + this.id + '</td><td>' + this.nome + '</td><td>' + FormatarData(this.inicio) +
            '</td><td>' + FormatarData(this.fim) + '</td><td>' + this.guid + '</td><td>\
                <a role="button" class="btn btn-warning" href="javascript:Alterar(' + this.id + ')">Alterar</a>\
                <a role="button" class="btn btn-danger" href="javascript:Excluir(' + this.id + ')">Excluir</a>\
                </td></tr>';
    });
    txt += '</tbody>';
    $("#tableQuestionarios").html(txt);
};

function ObterQuestionarios() {
    $("#divLoading").show(300);
    $.getJSON("/Questionario/ObterPorUsuario/", function (data) {
        PreencherTabela(data);
    });
    $("#divLoading").hide(300);
};

$(document).ready(function () {
    ObterQuestionarios();
});

$("#btnPesquisar").click(function () {
    if ($("#txtPalavraChave").val() == "") {
        ObterQuestionarios();
    }
    else {
        $("#divLoading").show(300);
        $.ajax({
            type: 'POST',
            url: '/Questionario/ObterPorPalavraChave',
            data: { Palavra: $("#txtPalavraChave").val() },
            success: function (result) {
                if (result != null && result.length > 0) {
                    PreencherTabela(result);
                }
                else {
                    bootbox.alert("Nenhum questionário encontrado.");
                }
                $("#divLoading").hide(300);
            },
            error: function (XMLHttpRequest, txtStatus, errorThrown) {
                alert("Status: " + txtStatus); alert("Error: " + errorThrown);
                $("#divLoading").hide(300);
            }
        });
    }
});

$("#btnConfirmar").click(function () {
    var msg = "";
    var id = $("#txtId").val();
    var titulo = $("#txtTitulo").val();
    var inicio = $("#txtDataInicio").val();
    var fim = $("#txtDataFim").val();
    var feedback = $("#txtFeedback").val();
    var guid = $("#txtGuid").val();

    if (titulo == "") {
        msg += "Por favor, informe um título para o questionário.<br />";
    }
    if (inicio == "") {
        msg += "Por favor, informe a data para início do questionário.<br />";
    }
    if (fim == "") {
        msg += "Por favor, informe a data para fechamento do questionário.<br />";
    }
    if (guid == "") {
        msg += "Por favor, informe a Guid (URL) para o questionário.<br />";
    }
    if (msg.length > 0) {
        Mensagem("divAlertaNovoQuestionario", msg);
    }
    else {
        $("#divLoading").show(300);
        $.ajax({
            type: 'POST',
            url: '/Questionario/Gravar',
            data: { Id: id, Nome: titulo, Inicio: inicio, Fim: fim, MsgFeedBack: feedback, Guid: guid },
            success: function (result) {
                $("#divLoading").hide(300);
                if (result.length > 0) {
                    Mensagem("divAlertaNovoQuestionario", result);
                }
                else {
                    LimparFormulario();
                    $.fancybox.close();
                    ObterQuestionarios();
                }
            },
            error: function (XMLHttpRequest, txtStatus, errorThrown) {
                alert("Status: " + txtStatus); alert("Error: " + errorThrown);
                $("#divLoading").hide(300);
            }
        });
    }
});

function Alterar(id) {
    $("#divLoading").show(300);
    $.ajax({
        type: 'POST',
        url: '/Questionario/Obter',
        data: { Id: id },
        success: function (result) {
            if (Object.keys(result).length > 0) {
                $("#divLoading").hide(300);
                $.fancybox.open({
                    src: '#formQuestionario',
                    type: 'inline'
                });
                $("#txtId").val(result.id);
                $("#txtTitulo").val(result.nome);
                $("#txtDataInicio").val(FormatarDataIso(result.inicio));
                $("#txtDataFim").val(FormatarDataIso(result.fim));
                $("#txtFeedback").val(result.msgFeedback);
                $("#txtGuid").val(result.guid);
            }
            else {
                $("#divLoading").hide(300);
            }
        },
        error: function (XMLHttpRequest, txtStatus, errorThrown) {
            alert("Status: " + txtStatus); alert("Error: " + errorThrown);
            $("#divLoading").hide(300);
        }
    });
};

function Excluir(id) {
    bootbox.confirm({
        message: "Confirma a exclusão deste registro?",
        buttons: {
            confirm: {
                label: 'Sim',
                className: 'btn-success'
            },
            cancel: {
                label: 'Não',
                className: 'btn-danger'
            }
        },
        callback: function (result) {
            if (result) {
                $.ajax({
                    type: 'POST',
                    url: '/Questionario/Excluir',
                    data: { Id: id },
                    success: function (result) {
                        if (result == "") {
                            ObterQuestionarios();
                        }
                        else {
                            Mensagem("divAlerta", result);
                        }
                    },
                    error: function (XMLHttpRequest, txtStatus, errorThrown) {
                        alert("Status: " + txtStatus); alert("Error: " + errorThrown);
                        $("#divLoading").hide(300);
                    }
                });
            }
        }
    });
};

function LimparFormulario() {
    $("#txtId").val("0");
    $("input[type='text']").val("");
    $("input[type='date']").val("0000-00-00");
    $("textarea").val("");
}